units/patch_intent.go

```diff
@@ -947,20 +948,34 @@ func (j *patchIntentProcessor) authAndFetchPRMergeBase(ctx context.Context, patc
 	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
 	defer cancel()
 
-	isMember, err := thirdparty.GithubUserInOrganization(ctx, githubOauthToken, requiredOrganization, githubUser)
-	if err != nil {
-		grip.Error(message.WrapError(err, message.Fields{
-			"job":          j.ID(),
-			"message":      "Failed to authenticate github PR",
-			"source":       "patch intents",
-			"creator":      githubUser,
-			"required_org": requiredOrganization,
-			"base_repo":    fmt.Sprintf("%s/%s", patchDoc.GithubPatchData.BaseOwner, patchDoc.GithubPatchData.BaseRepo),
-			"head_repo":    fmt.Sprintf("%s/%s", patchDoc.GithubPatchData.HeadOwner, patchDoc.GithubPatchData.HeadRepo),
-			"pr_number":    patchDoc.GithubPatchData.PRNumber,
-		}))
-		return false, err
-
+	isMember := false
+	var err error
+	// Github Dependabot patches should be automatically authorized.
+	if githubUser == githubDependabotUser {
+		grip.Info(message.Fields{
+			"job":       j.ID(),
+			"message":   fmt.Sprintf("authorizing patch from %s", githubDependabotUser),
+			"source":    "patch intents",
+			"base_repo": fmt.Sprintf("%s/%s", patchDoc.GithubPatchData.BaseOwner, patchDoc.GithubPatchData.BaseRepo),
+			"head_repo": fmt.Sprintf("%s/%s", patchDoc.GithubPatchData.HeadOwner, patchDoc.GithubPatchData.HeadRepo),
+			"pr_number": patchDoc.GithubPatchData.PRNumber,
+		})
+		isMember = true
+	} else {
+		isMember, err = thirdparty.GithubUserInOrganization(ctx, githubOauthToken, requiredOrganization, githubUser)
+		if err != nil {
+			grip.Error(message.WrapError(err, message.Fields{
+				"job":          j.ID(),
+				"message":      "Failed to authenticate github PR",
```

Comment:
should we update the error message here too?

Reply:
Update how?

Reply:
Ah i wasn't sure if we should follow kim's error style here saying something like "authenticating github pr"

Reply:
Oh gotcha -- this is for splunk so it's different; it's okay for messages to be full sentences.